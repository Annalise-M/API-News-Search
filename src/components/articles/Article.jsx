import React from 'react';
import PropTypes from 'prop-types';
import styles from './Article.css';

const Article = ({ title, author, description }) => (
  <figure className={styles.Article}>
    <p>{title}</p>
    <p>{author}</p>
    <p>{description}</p>

  </figure>
);

Article.propTypes = {
  title: PropTypes.string.isRequired,
  author: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired
};

export default Article;
