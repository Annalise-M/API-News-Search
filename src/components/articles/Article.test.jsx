import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Article from './Article';

describe('Article component', () => {
  afterEach(() => cleanup());
  it('renders Article', () => {
    // eslint-disable-next-line max-len
    const { asFragment } = render(<Article title="test" author="test" description="test" />);
   
    expect(asFragment()).toMatchSnapshot();
  });
});
