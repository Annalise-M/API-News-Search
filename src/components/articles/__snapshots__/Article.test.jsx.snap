// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article component renders Article 1`] = `
<DocumentFragment>
  <figure
    class="Article"
  >
    <p>
      test
    </p>
    <p>
      test
    </p>
    <p>
      test
    </p>
  </figure>
</DocumentFragment>
`;
